apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_permission.exs

```diff
@@ -0,0 +1,16 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_permissions, primary_key: false) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource, references(:resources, on_delete: :nothing, column: :name, type: :string)
+      add :permission, references(:permissions, on_delete: :nothing, column: :name, type: :string)
+
+      timestamps
+    end
+    create index(:roles_permissions, [:role_id])
```

Comment:
Yeah, a lot of these indexes were just sort of there through auto generated code, or me blindly copying the code in some cases. When I originally brought up the indexes in some other comment you were confirming my suspicions that fewer are better and it really needs to be tailored to the use of the table.

The only reason I'm doing unique here, is because I can't do a composite key at the moment through ecto and these should be unique. In ecto 2.0 I can declare these as primary keys for each field and drop the unique index and let postgress handle it via the primary key and let it manage the index.